src/test/java/com/connexta/search/index/IndexServiceImplTest.java

```diff
@@ -6,239 +6,175 @@
-    verifyNoMoreInteractions(mockSearchManager);
-  }
+  final BodyContentExtractor mockBodyContentExtractor = mock(BodyContentExtractor.class);
+  final IndexRepository mockIndexRepository = mock(IndexRepository.class);
+  final IonResourceLoader mockGetInputStream = mock(IonResourceLoader.class);
 
-  @ParameterizedTest
-  @ValueSource(ints = {400, 401, 403, 500, 501})
-  void testIrmUriReturnsNotOkStatusCode(final int code) throws Exception {
-    final String datasetId = "00067360b70e4acfab561fe593ad3f7a";
+  final IndexService indexService =
+      new IndexServiceImpl(mockIndexRepository, mockGetInputStream, mockBodyContentExtractor);
 
-    final SearchException thrown;
-    try (final MockWebServer storeMockWebServer = new MockWebServer()) {
-      storeMockWebServer.start();
-      storeMockWebServer.enqueue(new MockResponse().setResponseCode(code));
-
-      thrown =
-          assertThrows(
-              SearchException.class,
-              () ->
-                  ((IndexService) new IndexServiceImpl(mockSearchManager, WebClient.create()))
-                      .index(
-                          datasetId,
-                          storeMockWebServer
-                              .url(String.format("/dataset/%s/irm", datasetId))
-                              .uri()));
-    }
-    assertThat(thrown.getStatus(), is(HttpStatus.BAD_REQUEST));
+  @AfterEach
+  void afterEach() {
+    reset(mockBodyContentExtractor, mockIndexRepository, mockGetInputStream);
   }
 
   @Test
-  void testWebClientException(
-      @Mock final WebClient mockWebClient,
-      @Mock final RequestHeadersUriSpec mockRequestHeadersUriSpec,
-      @Mock final RequestHeadersSpec mockRequestHeadersSpec,
-      @Mock final ResponseSpec mockResponseSpec,
-      @Mock final ResponseSpec mockResponseSpec2,
-      @Mock final Mono<Resource> mockMono)
-      throws Exception {
-    final String datasetId = "00067360b70e4acfab561fe593ad3f7a";
-    final URI irmUri = new URI(String.format("http://store:9041/dataset/%s/irm", datasetId));
+  void testIndex() throws IOException, TikaException {
+    doReturn(IOUtils.toInputStream("asdf", StandardCharsets.UTF_8))
+        .when(mockGetInputStream)
+        .get(anyString());
+    doReturn("fileContents").when(mockBodyContentExtractor).extractText(any(InputStream.class));
 
-    when(mockWebClient.get()).thenReturn(mockRequestHeadersUriSpec);
-    when(mockRequestHeadersUriSpec.uri(irmUri)).thenReturn(mockRequestHeadersSpec);
-    when(mockRequestHeadersSpec.retrieve()).thenReturn(mockResponseSpec);
-    when(mockResponseSpec.onStatus(any(Predicate.class), any(Function.class)))
-        .thenReturn(mockResponseSpec2);
-    when(mockResponseSpec2.bodyToMono(Resource.class)).thenReturn(mockMono);
-
-    final RuntimeException runtimeException = new RuntimeException();
-    when(mockMono.block()).thenThrow(runtimeException);
+    indexService.index(Index.builder().id(DATASET_ID).fileUrl(FILE_URL).irmUrl(IRM_URL).build());
+  }
```

Comment:
❗️This test should verify 

- the correct URL was used in the loader
- the inputstream from the loader is the one used by the contentextractor
- the correct index was saved

```java
@Test
void testIndex(@Mock final InputStream mockInputStream) throws IOException, TikaException {
  doReturn(mockInputStream)
      .when(mockIonResourceLoader)
      .get(FILE_URL);
  doReturn("fileContents").when(mockBodyContentExtractor).extractText(mockInputStream);
  when(mockIndexRepository.existsById(datasetId)).thenReturn(false);

  indexService.index(Index.builder().id(DATASET_ID).fileUrl(FILE_URL).irmUrl(IRM_URL).build());

  verify(mockIndexRepository).save(TODO);
}
```